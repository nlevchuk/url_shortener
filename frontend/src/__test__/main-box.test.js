import MainBox, { OriginalLink } from '../main-box';
import Input from '../input';
import Button, { ShortenButton, CopyButton, ResetButton } from '../buttons';

describe('<MainBox />', () => {
  const url = 'https://heroku.com';
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<MainBox />);
  });

  it('should display shorten button', () => {
    expect(wrapper.find(ShortenButton)).toExist();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();

    wrapper.find(Input).simulate('change', { target: { value: url }});
    wrapper.find(ShortenButton).simulate('click');

    expect(wrapper).toMatchSnapshot();
  });

  it('should call alert when press on shorten button with empty input', () => {
    window.alert = jest.fn();
    expect(wrapper.find(Input)).not.toHaveValue();
    expect(window.alert).not.toHaveBeenCalled();

    // console.log(wrapper.debug());
    wrapper.find(ShortenButton).simulate('click');

    expect(window.alert).toHaveBeenCalled();
  });

  it('should display copy button', () => {
    expect(wrapper.find(CopyButton)).not.toExist();

    wrapper.find(Input).simulate('change', { target: { value: url }});
    wrapper.find(ShortenButton).simulate('click');

    expect(wrapper.find(CopyButton)).toExist();
  });

  it('should display reset button', () => {
    expect(wrapper.find(ResetButton)).not.toExist();

    wrapper.find(Input).simulate('change', { target: { value: url }});
    wrapper.find(ShortenButton).simulate('click');

    expect(wrapper.find(ResetButton)).toExist();
  });

  it('should display original link box', () => {
    expect(wrapper.find(OriginalLink)).not.toExist();

    wrapper.find(Input).simulate('change', { target: { value: url }});
    wrapper.find(ShortenButton).simulate('click');

    expect(wrapper.find(OriginalLink)).toExist();
  });

  it('should call alert when press on copy button', () => {
    window.alert = jest.fn();
    expect(window.alert).not.toHaveBeenCalled();

    wrapper.find(Input).simulate('change', { target: { value: url }});
    wrapper.find(ShortenButton).simulate('click');

    expect(window.alert).not.toHaveBeenCalled();

    wrapper.find(CopyButton).simulate('click');

    expect(window.alert).toHaveBeenCalled();
  });

  it('should reset everything when press on reset button', () => {
    wrapper.find(Input).simulate('change', { target: { value: url }});
    wrapper.find(ShortenButton).simulate('click');

    wrapper.find(ResetButton).simulate('click');

    expect(wrapper.find(Input)).not.toHaveValue();
    expect(wrapper.find(ShortenButton)).toExist();
    expect(wrapper.find(ResetButton)).not.toExist();
    expect(wrapper.find(CopyButton)).not.toExist();
    expect(wrapper.find(OriginalLink)).not.toExist();
  });
});
