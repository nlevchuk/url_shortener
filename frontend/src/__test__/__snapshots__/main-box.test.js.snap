// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainBox /> should render correctly 1`] = `
<MainBox>
  <main>
    <div
      id="container"
    >
      <div
        className="row"
        id="form-container"
      >
        <div
          id="input-item"
        >
          <Input
            onChange={[Function]}
            readOnly={false}
            url=""
          >
            <input
              onChange={[Function]}
              placeholder="Paste URL here"
              type="text"
              value=""
            />
          </Input>
        </div>
        <div
          id="button-item"
        >
          <ShortenButton
            onClick={[Function]}
          >
            <Button
              onClick={[Function]}
              value="SHORTEN"
            >
              <button
                onClick={[Function]}
              >
                SHORTEN
              </button>
            </Button>
          </ShortenButton>
        </div>
      </div>
    </div>
  </main>
</MainBox>
`;

exports[`<MainBox /> should render correctly 2`] = `
<MainBox>
  <main>
    <div
      id="container"
    >
      <div
        className="row"
        id="form-container"
      >
        <div
          id="input-item"
        >
          <Input
            onChange={[Function]}
            readOnly={true}
            url="blahblah"
          >
            <input
              placeholder="Paste URL here"
              readOnly={true}
              type="text"
              value="blahblah"
            />
          </Input>
          <ResetButton
            onClick={[Function]}
          >
            <Button
              id="reset-button"
              onClick={[Function]}
              value="X"
            >
              <button
                id="reset-button"
                onClick={[Function]}
              >
                X
              </button>
            </Button>
          </ResetButton>
        </div>
        <div
          id="button-item"
        >
          <CopyButton
            onClick={[Function]}
          >
            <Button
              onClick={[Function]}
              value="COPY"
            >
              <button
                onClick={[Function]}
              >
                COPY
              </button>
            </Button>
          </CopyButton>
        </div>
      </div>
      <OriginalLink
        href="https://heroku.com"
      >
        <div
          className="row"
          id="original-link"
        >
          <div>
            <a
              href="https://heroku.com"
            >
              https://heroku.com
            </a>
          </div>
          <span>
            shortening
          </span>
        </div>
      </OriginalLink>
    </div>
  </main>
</MainBox>
`;
